.github/workflows/test.yml

```diff
@@ -0,0 +1,54 @@
+name: "Run Tests for all samples"
+
+on:
+   push:
+    branches: [ "master" ]
+   pull_request:
+    branches: [ "master" ]
+
+jobs:
+  prepare_list:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-matrix
+        run: |
+          echo "matrix=[$(find . -mindepth 2 -type f -name 'Makefile' ! -path './sample-archive/*' | sed 's|/Makefile||' | awk '{printf "\"%s\",", $1}' | sed 's/,$//')]" >> $GITHUB_OUTPUT
+      - run: echo "matrix=[$(find . -mindepth 2 -type f -name 'Makefile' ! -path './sample-archive/*' | sed 's|/Makefile||' | awk '{printf "\"%s\",", $1}' | sed 's/,$//')]"
```

Comment:
This is a pretty dangerous approach, because it does not limit the amount of parallel runners at all. GitHub organizations have a limit of 20 concurrent Linux runners for free in public repos: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
This means that it would basically block all GitHub action executions for the time it takes to execute the pipeline for the whole `localstack-samples` org.
Since these samples are executed somewhat fast, it can be a conscious decision to leave it like this, but I would propose to at least limit the amount of concurrent runners to less than 20 (maybe 5-10) to allow other pipelines in the org to be scheduled at the same time.
You can implement that super-easily with the `max-parallel` setting in the matrix config: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#defining-the-maximum-number-of-concurrent-jobs